File: native/swift/Sources/BlocksRuntime/OIDC/TokenStore.swift
import Foundation

/// Persists OIDC tokens (access, refresh, expires_at, PKCE state).
public protocol TokenStore: Sendable {
    func get(_ key: String) async -> String?
    func set(_ key: String, value: String) async
    func delete(_ key: String) async
}

/// In-memory token store. Tokens are lost when the process exits.
public actor InMemoryTokenStore: TokenStore {
    private var storage: [String: String] = [:]

    public init() {}

    public func get(_ key: String) async -> String? {
        storage[key]
    }

    public func set(_ key: String, value: String) async {
        storage[key] = value
    }

    public func delete(_ key: String) async {
        storage.removeValue(forKey: key)
    }
}

File: native/dart/packages/blocks_runtime/lib/src/token_store.dart
/// Abstract interface for persisting OIDC tokens.
abstract class TokenStore {
  Future<String?> get(String key);
  Future<void> set(String key, String value);
  Future<void> delete(String key);
}

/// In-memory token store — tokens are lost when the process exits.
class InMemoryTokenStore implements TokenStore {
  final _map = <String, String>{};

  @override
  Future<String?> get(String key) async => _map[key];

  @override
  Future<void> set(String key, String value) async => _map[key] = value;

  @override
  Future<void> delete(String key) async => _map.remove(key);
}
